cluster/src/main/scala/org/apache/spark/sql/streaming/SnappyStreamingContextFactory.scala

```diff
@@ -61,13 +56,18 @@ class SnappyStreamingContextFactory extends SparkContextFactory {
     new SnappyStreamingContext(LeadImpl.getInitializingSparkContext,
       Milliseconds(interval)) with ContextLike {
 
+      private val addedJars = scala.collection.mutable.ArrayBuffer.empty[String]
```

Comment:
Not really, its usage we have suggested makes it singleton like. You can always do new SnappyStreamingContext.  Job-server ( our changes) usage this to create s new SnappyStreamingContext every time a stream job is submitted.  Only when one starts a streaming context it validates if other context is running or not.